src/components/layoutcomponents/Header.astro

```diff
@@ -143,20 +143,20 @@ const { navigation } = Astro.props;
 
     const setInitialDotState = () => {
       if (activeMainNavLink) {
-        handleDotPosition(activeMainNavLink, mainNav);
+        handleDotPosition(activeMainNavLink, mainNav!);
         showDotIndicator();
       } else {
         hideDotIndicator();
       }
     };
 
     const mainNavigationResizeObserver = new ResizeObserver((entries) => {
-      for (let entry of entries) {
+      for (let i = 0; i < entries.length; i++) {
```

Comment:
isn't this the same? Other than the index variable which we dont use

Reply:
We don't use entry in the function. We only use the index.